qcodes/instrument_drivers/tektronix/keithley_7510.py

```diff
@@ -0,0 +1,241 @@
+        unit = self.function_modes[self._proper_function]["unit"]
+
+        self.function = self.parent.sense_function
+
+        self.add_parameter(
+            self._proper_function,
+            get_cmd=":MEASure?",
+            get_parser=float,
+            unit=unit,
+        )
+
+        self.add_parameter(
+            "auto_range",
+            get_cmd=f":SENSe:{self._proper_function}:RANGe:AUTO?",
+            set_cmd=f":SENSe:{self._proper_function}:RANGe:AUTO {{}}",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0")
+        )
+
+        self.add_parameter(
+            "range",
+            get_cmd=f":SENSe:{self._proper_function}:RANGe?",
+            set_cmd=f":SENSe:{self._proper_function}:RANGe {{}}",
+            vals=range_vals,
+            get_parser=float,
+            unit=unit
+        )
+
+        self.add_parameter(
+            "nplc",
+            get_cmd=f":SENSe:{self._proper_function}:NPLCycles?",
+            set_cmd=f":SENSe:{self._proper_function}:NPLCycles {{}}",
+            vals=Numbers(0.01, 10),
+            get_parser=float,
+        )
+
+        self.add_parameter(
+            "auto_delay",
+            get_cmd=f":SENSe:{self._proper_function}:DELay:AUTO?",
+            set_cmd=f":SENSe:{self._proper_function}:DELay:AUTO {{}}",
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF")
+        )
+
+        self.add_parameter(
+            'user_number',
+            get_cmd=None,
+            set_cmd=None,
+            vals=Ints(1, 5)
+        )
+
+        self.add_parameter(
+            "user_delay",
+            get_cmd=self._get_user_delay,
+            set_cmd=self._set_user_delay,
+            vals=Numbers(0, 1e4),
+            unit='second'
+        )
+
+        self.add_parameter(
+            "auto_zero",
+            get_cmd=f":SENSe:{self._proper_function}:AZERo?",
+            set_cmd=f":SENSe:{self._proper_function}:AZERo {{}}",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0")
+        )
+
+        self.add_parameter(
+            "auto_zero_once",
+            set_cmd=f":SENSe:AZERo:ONCE",
+        )
+
+        self.add_parameter(
+            "average",
+            get_cmd=f":SENSe:{self._proper_function}:AVERage?",
+            set_cmd=f":SENSe:{self._proper_function}:AVERage {{}}",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0")
+        )
+
+        self.add_parameter(
+            "average_count",
+            get_cmd=f":SENSe:{self._proper_function}:AVERage:COUNt?",
+            set_cmd=f":SENSe:{self._proper_function}:AVERage:COUNt {{}}",
+            vals=Numbers(1, 100)
+        )
+
+        self.add_parameter(
+            "average_type",
+            get_cmd=f":SENSe:{self._proper_function}:AVERage:TCONtrol?",
+            set_cmd=f":SENSe:{self._proper_function}:AVERage:TCONtrol {{}}",
+            vals=Enum('REP', 'rep', 'MOV', 'mov')
+        )
```

Comment:
Could you please add short doc-strings to the parameters via `docstring` attribute?

Reply:
docstrings have been added to the parameters added.